core/imageroot/usr/local/agent/pypkg/agent/__init__.py

```diff
@@ -412,3 +412,29 @@ def remove_public_service(name):
         data={'service': name}
     )
     return response['exit_code'] == 0
+
+def get_smarthost_settings():
+    rdb = redis_connect()
```

Comment:
Like I did in a similar function (pr #271), I'd prefer to pass the Redis connection object as function argument (`rdb`) because the code has better ways to check for errors and customize the connection parameters according to its needs.

```suggestion
def get_smarthost_settings(rdb):
```